ext/js/dictionary/dictionary-importer.js

```diff
@@ -691,18 +691,19 @@ export class DictionaryImporter {
     _getArchiveFiles(fileMap, queryDetails) {
         /** @type {import('dictionary-importer').QueryResult} */
         const results = new Map();
-        for (const [name, value] of fileMap.entries()) {
-            for (const [fileType, fileNameFormat] of queryDetails.entries()) {
-                let entries = results.get(fileType);
-                if (typeof entries === 'undefined') {
-                    entries = [];
-                    results.set(fileType, entries);
-                }
 
-                if (fileNameFormat.test(name)) {
-                    entries.push(value);
-                    break;
-                }
+        for (const [fileType] of queryDetails) {
+            results.set(fileType, []);
+        }
+
+        for (const [fileName, fileEntry] of fileMap.entries()) {
+            for (const [fileType, fileNameFormat] of queryDetails) {
+                if (!fileNameFormat.test(fileName)) { continue; }
+                const entries = results.get(fileType);
+
+                // @ts-expect-error - entries should be initialized
```

Comment:
Avoid using `@ts` pragmas if not absolutely needed; just do an extra if undefined check.